commit 92269e0178bc14c6cdaefa567652c5a6de832e16
Author: Steve Muir <muir@fb.com>
Date:   Wed Jul 27 14:19:37 2016 -0700

    yummy:patchX:tests--don-t-test-hostname-if-it-looks-like-an-id128.patch

tests: don't test hostname if it looks like an id128

    
    The condition tests for hostname will fail if hostname looks like an id128.
    The test function attempts to convert hostname to an id128, and if that
    succeeds compare it to the machine ID (presumably because the 'hostname'
    condition test is overloaded to also test machine ID). That will typically
    fail, and unfortunately the 'mock' utility generates a random hostname that
    happens to have the same format as an id128, thus causing a test failure.

diff --git a/src/test/test-condition.c b/src/test/test-condition.c
index 987862f..4ef61eb 100644
--- a/src/test/test-condition.c
+++ b/src/test/test-condition.c
@@ -25,6 +25,7 @@
 #include "audit-util.h"
 #include "condition.h"
 #include "hostname-util.h"
+#include "id128-util.h"
 #include "ima-util.h"
 #include "log.h"
 #include "macro.h"
@@ -142,9 +143,14 @@ static void test_condition_test_host(void) {
         hostname = gethostname_malloc();
         assert_se(hostname);
 
-        condition = condition_new(CONDITION_HOST, hostname, false, false);
-        assert_se(condition_test(condition));
-        condition_free(condition);
+        /* if hostname looks like an id128 then skip testing it */
+        if (id128_is_valid(hostname)) {
+                log_notice("hostname is an id128, skipping test");
+        } else {
+                condition = condition_new(CONDITION_HOST, hostname, false, false);
+                assert_se(condition_test(condition));
+                condition_free(condition);
+        }
 }
 
 static void test_condition_test_architecture(void) {
